src/selectors/identity.js

```diff
@@ -21,28 +21,44 @@ export const selectClaimCategoriesObj = (state) => state.identity.getIn(['person
 
 export const selectShowEmptyClaimCategories = (state) => state.identity.getIn(['personalInformation', 'showEmptyClaimCategories'], false);
 
+export const selectClaims = (state) => state.identity.getIn(['personalInformation', 'claims', 'byId'], IMap());
+
 export const selectClaimCategoriesToDisplay = createSelector(
-  [selectClaimCategories, selectShowEmptyClaimCategories],
-  (claimCategories, showEmptyClaimCategories) => {
+  [selectClaimCategories, selectShowEmptyClaimCategories, selectClaims],
+  (claimCategories, showEmptyClaimCategories, claims) => {
     if (!showEmptyClaimCategories) {
-      return claimCategories.filter((claimCategory) => claimCategory.get('claims', IList()).size > 0);
+      return claimCategories.filter((claimCategory) => claims.some((claim) => claim.get('categoryId', '') === claimCategory.get('id', '')));
     }
     return claimCategories;
   },
 );
 
-const selectClaims = (state) => state.identity.getIn(['personalInformation', 'claims', 'byId'], IMap());
+export const selectClaimsObj = (state) => state.identity.getIn(['personalInformation', 'claims'], IMap());
 
 const selectActiveClaimCategoryId = (state) => state.identity.getIn(['personalInformation', 'activeClaimCategoryId'], '');
 
+export const selectShowHiddenClaims = (state) => state.identity.getIn(['personalInformation', 'showHiddenClaims'], false);
+
+export const selectSelectedClaims = (state) => state.identity.getIn(['personalInformation', 'selectedClaims'], IList());
+
 const selectActiveCategory = createSelector(
   [selectActiveClaimCategoryId, selectClaimCategories],
   (claimCategoryId, claimCategories) => claimCategories.filter((claimCategory) => claimCategory.get('id', '') === claimCategoryId),
 );
 
 export const selectClaimsByCategoryId = createSelector(
-  [selectActiveClaimCategoryId, selectActiveCategory, selectClaims],
-  (activeCategoryId, activeCategory, claims) => claims.filter((claim) => activeCategory.getIn([activeCategoryId, 'claims'], IList()).indexOf(claim.get('id', '')) > -1),
+  [selectActiveClaimCategoryId, selectClaims],
+  (activeCategoryId, claims) => claims.filter((claim) => claim.get('categoryId', '') === activeCategoryId)
+);
+
+export const selectClaimsToDisplay = createSelector(
+  [selectClaimsByCategoryId, selectShowHiddenClaims],
+  (selectedClaims, showHiddenClaims) => {
+    if (!showHiddenClaims) {
```

Comment:
Why the reverse, you are just adding an additional operation
if(showHiddenClaims) {
  return selectedClaims
}
....